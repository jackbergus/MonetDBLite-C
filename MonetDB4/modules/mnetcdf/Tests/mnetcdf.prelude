module("unix");
module("mnetcdf");

VAR TSTTRGDIR := getenv("TSTTRGDIR");

