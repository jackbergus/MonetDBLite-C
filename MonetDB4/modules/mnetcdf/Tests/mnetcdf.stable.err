stderr of test 'mnetcdf` in directory 'modules/mnetcdf` itself:


# 12:29:53 >  
# 12:29:53 >  Mtimeout -timeout 60 Mserver "--config=/data/home/alexb/local/Monet/soft/etc/monet.conf" --debug=10 --set "monet_mod_path=/data/home/alexb/local/Monet/soft/lib/MonetDB" --set "gdk_dbfarm=/data/home/alexb/local/Monet/soft/var/MonetDB/dbfarm" --set "sql_logdir=/data/home/alexb/local/Monet/soft/var/MonetDB/log" --set mapi_port=43054 --set sql_port=58740 --set monet_prompt= --trace --dbname=mTests_src_modules_contrib mnetcdf.prelude < mnetcdf.milM
# 12:29:53 >  


# 12:29:53 >  
# 12:29:53 >  Done.
# 12:29:53 >  

