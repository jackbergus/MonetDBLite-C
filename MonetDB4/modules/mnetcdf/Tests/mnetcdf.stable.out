stdout of test 'mnetcdf` in directory 'modules/mnetcdf` itself:


# 16:35:14 >  
# 16:35:14 >  Mtimeout -timeout 60 Mserver "--config=/data/home/alexb/local/Monet/soft/etc/MonetDB.conf" --debug=10 --set "monet_mod_path=/data/home/alexb/local/Monet/soft/lib/MonetDB" --set "gdk_dbfarm=/data/home/alexb/local/Monet/soft/var/MonetDB/dbfarm" --set "sql_logdir=/data/home/alexb/local/Monet/soft/var/MonetDB/log" --set mapi_port=44200 --set sql_port=59829 --set monet_prompt= --trace --dbname=mTests_src_modules_contrib mnetcdf.prelude < mnetcdf.milM
# 16:35:14 >  

# Monet Database Server V4.4.1
# Copyright (c) 1993-2004, CWI. All rights reserved.
# Compiled for x86_64-unknown-linux-gnu/64bit; dynamically linked.
# Visit http://monetdb.cwi.nl/ for further information.
PROC test_mnetcdf() : void := {
    var data,idx1,idx2,ncid,variables,vinfo,input;
    data := new(void,int);
    data.insert(nil,rand());
    data.insert(nil,rand());
    data.insert(nil,rand());
    data.insert(nil,rand());
    seqbase(data,0@0);
    data := [flt](data);

    idx1 := new(void,int);
    idx1.insert(nil,0);
    idx1.insert(nil,0);
    idx1.insert(nil,1);
    idx1.insert(nil,1);
    seqbase(idx1,0@0);

    idx2 := new(void,int);
    idx2.insert(nil,0);
    idx2.insert(nil,1);
    idx2.insert(nil,0);
    idx2.insert(nil,1);
    seqbase(idx2,0@0);

    print(data,idx1,idx2);

    ncid := nc_create(TSTTRGDIR + "/testdata.nc");
    nc_export(ncid,"TEST_DATA",data,idx1,idx2);
    nc_export_fast(ncid,"TEST_DATA_fast",data,2,2);
    nc_close(ncid);
    
    ncid      := nc_open(TSTTRGDIR + "/testdata.nc");
    variables := nc_list(ncid);
    print(variables);
    vinfo     := nc_describe(ncid,"TEST_DATA");
    print(vinfo);
    input     := nc_import(ncid,"TEST_DATA");
    print(input);
    vinfo     := nc_describe(ncid,"TEST_DATA_fast");
    print(vinfo);
    input     := nc_import(ncid,"TEST_DATA_fast");
    print(input);
    nc_close(ncid);
}
test_mnetcdf();
#-----------------------------------------#
# h	tmp_29		tmp_28	tmp_30	  # name
# void	flt		int	int	  # type
#-----------------------------------------#
[ 0@0,	  1.80428941e+09, 0,	  0	  ]
[ 1@0,	  846930880,	  0,	  1	  ]
[ 2@0,	  1.6816928e+09,  1,	  0	  ]
[ 3@0,	  1.71463693e+09, 1,	  1	  ]
#---------------------------------#
# h	tmp_31			  # name
# void	str			  # type
#---------------------------------#
[ 0@0,	  "TEST_DATA"		  ]
[ 1@0,	  "TEST_DATA_fast"	  ]
#-------------------------------------------------------------------------#
# h		tmp_32							  # name
# str		str							  # type
#-------------------------------------------------------------------------#
[ "name",	  "TEST_DATA"						  ]
[ "id",		  "0"							  ]
[ "netCDF type",  "FLOAT"						  ]
[ "MonetDB type", "flt"							  ]
[ "dimensions",	  "2"							  ]
[ "shape",	  "2,2"							  ]
[ "size",	  "4"							  ]
[ "attributes",	  "1"							  ]
[ "_FillValue",	  "-340282346638528859811704183484516925440.000000"	  ]
#-------------------------#
# h	tmp_33		  # name
# void	flt		  # type
#-------------------------#
[ 0@0,	  1.80428941e+09  ]
[ 1@0,	  846930880	  ]
[ 2@0,	  1.6816928e+09	  ]
[ 3@0,	  1.71463693e+09  ]
#-------------------------------------------------------------------------#
# h		tmp_34							  # name
# str		str							  # type
#-------------------------------------------------------------------------#
[ "name",	  "TEST_DATA_fast"					  ]
[ "id",		  "1"							  ]
[ "netCDF type",  "FLOAT"						  ]
[ "MonetDB type", "flt"							  ]
[ "dimensions",	  "2"							  ]
[ "shape",	  "2,2"							  ]
[ "size",	  "4"							  ]
[ "attributes",	  "1"							  ]
[ "_FillValue",	  "-340282346638528859811704183484516925440.000000"	  ]
#-------------------------#
# h	tmp_32		  # name
# void	flt		  # type
#-------------------------#
[ 0@0,	  1.80428941e+09  ]
[ 1@0,	  846930880	  ]
[ 2@0,	  1.6816928e+09	  ]
[ 3@0,	  1.71463693e+09  ]
quit();

# 16:35:14 >  
# 16:35:14 >  Done.
# 16:35:14 >  

